org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardPageBranchAndPackage.java

```diff
@@ -381,6 +402,34 @@ private void retrieveDependentManifests(ApackParameters apackParameters, final H
 		}
 	}
 
+	//A backend query to fetch the modified objects for current repository
+	private void fetchModifiedObjects() {
```

Comment:
I agree that to reduce redundancy, this logic should be placed centrally, as it can be used in 3 classes.
So, I have placed this in method `fetchAndExtractModifiedObjectsToPull`, `RepositoryUtil` class in `org.abapgit.adt.ui.internal.util` package.